packages/type-safe-api/src/project/codegen/runtime/generated-typescript-runtime-project.ts

```diff
@@ -123,6 +123,20 @@ export class GeneratedTypescriptRuntimeProject extends TypeScriptProject {
       ".openapi-generator"
     );
 
+    // Add compilation to ESM as well
+    const compileESMTask = this.addTask("compile-esm");
+    compileESMTask.exec("tsc -p tsconfig.esm.json");
+    new JsonFile(this, "tsconfig.esm.json", {
+      obj: {
+        extends: "./tsconfig.json",
+        compilerOptions: {
+          module: "esnext",
+          moduleResolution: "nodenext",
+          outDir: "lib/esm",
+        },
+      },
+    });
```

Comment:
how do I conditionally add this generation to the task? This file and the fields in `package.json` need to only be added when this task is run, right?

Reply:
We probably should be extending `tsconfig.dev.json `as this is what we actually use to compile.

There is no harm in having this file generated and the task instrumented all the time.

also, you probably want to execute this during compile, to do that you can add this after your task definition:

`this.compileTask.spawn(compileESMTask);`

Reply:
I think we want to keep a `tsconfig.esm.json` that's separate so we can build both cjs and esm :)